### PR TITLE
Link to Python source code

### PR DESCRIPTION
**Pull request by leonlan**
This PR partially addresses #345. It implements a custom `linkcode_resolve` function that generates a URL to the Github source code for documented objects. It currently only works for native Python objects.

### CODE REVIEW

File: docs/source/conf.py [hunk 1]
```diff
@@ -26,6 +30,62 @@
...
+    rel_path = os.path.relpath(source_file, parent_dir_path)
+    start_line_no = inspect.getsourcelines(obj)[1]
+
+    # Get the current git revision. If it cannot be determined, return "main".
+    try:
+        cmd = "git rev-parse --short HEAD".split()
+        output = subprocess.check_output(cmd).strip()
+        revision = output.decode("utf-8")
+    except (subprocess.CalledProcessError, OSError):
+        revision = "main"
```
Comment (leonlan) [inline docs/source/conf.py:83]: Doesn't work yet for RTD.

Comment (leonlan) [reply docs/source/conf.py:83]: Maybe need to use [sphinx-git](https://sphinx-git.readthedocs.io/en/latest/) for this.

Comment (leonlan) [reply docs/source/conf.py]: https://readthedocs.org/api/v2/build/21846816.txt

CTRL-f "git revision" shows that the function does identify the correct git revision. But why then do the links not get resolved correctly?

Comment (leonlan) [reply docs/source/conf.py]: Found the problem: the URLs are wrong 😅 I ran `poetry run make docs` from inside `docs/`, so the paths are different than when it's called from the root directory.

```
https:///github.com/PyVRP/PyVRP/blob/4bf717d/../pyvrp/plotting/plot_coordinates.py#L9
https:///github.com/PyVRP/PyVRP/blob/4bf717d/../pyvrp/plotting/plot_demands.py#L9
```

**Comment by leonlan** [comment]
This seems to work now. @N-Wouda could you review this?

**Comment by N-Wouda** [comment]
I'll try to have a look at this tomorrow!

**Comment by N-Wouda** [comment]
Is the `REVISION` value something the `linkcode` extension requires (like the function definition), or is it something we use to form the URL?

**Comment by leonlan** [comment]
> Is the `REVISION` value something the `linkcode` extension requires (like the function definition), or is it something we use to form the URL?

The latter!

**Review by N-Wouda — APPROVED** [review]
LGTM! Let's wait for RTD to complete, and if that looks alright I think we can merge this in.

**Pull request merged**